tests/test_tekrsa_sigan.py

```diff
@@ -11,13 +22,189 @@ class TestTekRSA:
+
+    def test_iq_bandwidth(self):
+        assert isinstance(self.rx.iq_bandwidth, (float, int))
+
+        for bw in self.CORRECT_ALLOWED_BW:
+            setattr(self.rx, "iq_bandwidth", bw)
+            assert self.rx.iq_bandwidth == bw
+            # Sample rate should update when bandwidth is set
+            assert self.rx.sample_rate == rsa_constants.IQSTREAM_BW_SR_MAP[bw]
+
+        with pytest.raises(ValueError):
+            # Requested bandwidth invalid
+            setattr(self.rx, "iq_bandwidth", max(self.CORRECT_ALLOWED_BW) + 1)
+
+    def test_frequency(self):
+        assert isinstance(self.rx.frequency, (float, int))
+
+        for cf in np.linspace(self.rx.min_frequency, self.rx.max_frequency, 10):
+            setattr(self.rx, "frequency", cf)
+            assert self.rx.frequency == cf
+
+    def test_reference_level(self):
+        assert isinstance(self.rx.reference_level, (float, int))
+
+        for rl in np.linspace(
+            self.rx.min_reference_level, self.rx.max_reference_level, 10
+        ):
+            setattr(self.rx, "reference_level", rl)
+            assert self.rx.reference_level == rl
+
+    def test_attenuation(self):
+        assert isinstance(self.rx.attenuation, (float, int))
+
+        for a in np.linspace(self.rx.min_attenuation, self.rx.max_attenuation, 10):
+            setattr(self.rx, "attenuation", a)
+            assert self.rx.attenuation == a
+
+        with pytest.raises(ValueError):
+            # Requested attenuation too high
+            setattr(self.rx, "attenuation", self.rx.max_attenuation + 1)
+
+        # Test handling for RSA without manual attenuator
+        old_dev_name = self.rx.device_name
+        setattr(self.rx, "device_name", "RSA306B")
+        assert self.rx.attenuation is None
+        setattr(self.rx, "attenuation", 50)
+        assert self.rx.attenuation is None
+        setattr(self.rx, "device_name", old_dev_name)
+
+    def test_preamp_enable(self):
+        assert isinstance(self.rx.preamp_enable, bool)
+
+        setattr(self.rx, "preamp_enable", False)
+        assert self.rx.preamp_enable == False
+        setattr(self.rx, "preamp_enable", True)
+        assert self.rx.preamp_enable == True
+
+        # Test handling for RSA without preamp
+        old_dev_name = self.rx.device_name
+        setattr(self.rx, "device_name", "RSA306B")
+        assert self.rx.preamp_enable is None
+        setattr(self.rx, "preamp_enable", False)
+        assert self.rx.preamp_enable is None
+        setattr(self.rx, "device_name", old_dev_name)
+
+    def test_acquire_samples_retry(self):
+        # Not enough retries = acquisition should fail
+        with pytest.raises(RuntimeError):
+            for i in range(TIMES_TO_FAIL):
+                _ = self.rx.acquire_time_domain_samples(
+                    100, retries=TIMES_TO_FAIL - 1, cal_adjust=False
+                )
+
+        # Now enough retries are available, should succeed immediately
```

Comment:
this comment is a little confusing. I think it is not doing any more retries after going through for loop right?

Reply:
I'm glad you pointed this out, because the test itself was not written as it should have been. I rewrote that test in 13be078464b65ac11ad46961cad9f045436063ca, so that it now checks that the retry logic in `acquire_time_domain_samples`.